Automate game state

alexrutar opened the pull request:
This WIP is to automate the creation of all the game state files.

alexrutar committed:
first code separation

This commit pulls the HTML content out of the markdown, and also fixed the
markdown content a bit

alexrutar committed:
clean up code

- Remove weird XHTML-style tags (/> and ;>), and other minor spacing fixes.
- fully defer script until end

alexrutar committed:
improve javascript

- simplify code to have less repetition
- rename some variables to be a bit more consistent
- change site save structure to use array instead of named elements

alexrutar committed:
improve javascript data handling

This commit improves the oath parser output format to be somewhat more
robust, by converting the suit array into a dict. This also updates the
suit count rendering to generate the entire table automatically in a
non-fragile way.

To customize the table order, change the order in the 'suits' array.

alexrutar committed:
rename oath-parser file and update README to be consistent

alexrutar commented on assets/js/oath-parser.js:
```diff
@@ -4,8 +4,9 @@ import * as other from './other.js'
-const worldSuitArray = [0,0,0,0,0,0]
-const dispSuitArray = [0,0,0,0,0,0]
+const suits = ['discord', 'hearth', 'nomad', 'arcane', 'order', 'beast']
+const worldSuitDict = {'discord': 0, 'hearth': 0, 'nomad': 0, 'arcane': 0, 'order': 0, 'beast': 0}
```
@AndBM You should take a look at the changes here for the future in case you are updating the JS again later

alexrutar committed:
make the quote and table spacing a bit cleaner

alexrutar committed:
some style improvements

- improve the spacing and alignment in player profiles
- make the savestring copy the entire text on-click

alexrutar committed:
separate data into game_data.js

Input the player-entered data can be input into game_data.js for more
consistency in the future. Also update README to include new
instructions.

alexrutar committed:
remove unnecessary tags and ids

alexrutar commented:
TODO:

- [ ] Convert sites to a table, and render them in a more robust way. Maybe worth adding the site images, as well?
- [ ] Render the `#player-profiles` table automatically. This will require deciding on a data format, adding the data to `assets/js/game_data.js`, and then generating the table with this information along with the info pulled from the savestring.
- [ ] Restructure site data output from `oath_parser.js` into the form `{cradle: [site, ...], ...}` rather than an index-sensitive array.
- [x] Change suit order to alphabetical, or match board order, etc. (just modify `suits` array).

Basically, the required data should be something like a dict with keys that are the house colors, e.g.
```js
"yellow": {
  "player": "Andreas",
  "name": "House of Dawn"
}
```

alexrutar commented:
Also, it would be nice if the table of player images were responsive: i.e. display the images in a single column if the page is sufficiently narrow, and in 2 columns otherwise.

alexrutar committed:
touch up script a bit

Reoder script to match the document order, and other minor cleanup
without changing the rendered DOM.

alexrutar commented:
The suit order can now be changed. Is there a "canonical order"? Perhaps alphabetical, or perhaps the order presented on the board itself.

AndBM commented:
>The suit order can now be changed. Is there a "canonical order"? Perhaps alphabetical, or perhaps the order presented on the board itself.

There is a canonical order in the LUA files, but it doesn't really mean anything. I would actually prefer to change the order to the order on the board: Discord, Arcane, Order, Hearth, Beast, Nomad.

alexrutar commented:
Also, what does the `save.name` variable mean? In the parser output.

alexrutar committed:
change suit order to match board

AndBM commented:
The `save.name` string is the chronicle name, in this case `"The Helichrysum Chronicle"`

alexrutar commented:
Cool, I can render this into the state as well automatically.

AndBM commented:
It already is, in default.html, the element with id "name".

alexrutar committed:
Render chronicle title from _config.yml

Also remove unused JS which rendered the title in some past version.

alexrutar commented:
That element no longer seems to exist? I must have deleted it somewhere, and I cannot remember when. It's missing already in the `main` branch.

Anyway I added it back now. Also, I realized it's possible to access site variables directly from the JS at render-time! This means all the configuration files don't need to be in `game_state.js` and can instead be put directly into `_config.yml`. This is really convenient from a packaging / organizational perspective.

Edit: I've realized this is a bad idea - see #11 for the proper way to do this.

alexrutar committed:
update readme and add houses and player titles to game data

alexrutar committed:
remove conspiracy

alexrutar committed:
convert sites into <ul>

alexrutar committed:
autogenerate site names

This also updates the save.sites data structure, and adds a regions
variable describing the valid keys.

alexrutar committed:
add <hr> at the end of the game state section

alexrutar committed:
convert person table to grid

alexrutar committed:
create player display function

alexrutar commented:
Alright all done now! The code is fully running - just the calls to render the player table is hard-coded. I'll leave it to you to edit this to get the correct calls out of the savestring.

alexrutar commented on assets/js/script.js:
```diff
@@ -30,6 +30,44 @@ const oathFancyName = {
+// prevGameList.forEach(
+//   player => document.getElementById("player-profiles").appendChild(createPlayerProfile(player.color, player.citizenStatus))
+// )
+// and don't forget to delete the contents of the "player-profiles" div
```
nevermind I already did this

AndBM committed:
Change house Red to Red Lotus

AndBM committed:
small chancellor name bug

Plus test if I can write to branch

AndBM committed:
updated byte parser and automated player list

alexrutar committed:
minor code formatting fixes
